internal/http/web/raffle.go

```diff
@@ -0,0 +1,203 @@
+package web
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/BoilerMake/bm-app/internal/models"
+	"github.com/BoilerMake/bm-app/pkg/flash"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+// getRaffle generates the page for users to enter a raffle
+func (h *Handler) getRaffle() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		session := h.getSession(r)
+
+		id, ok := session.Values["ID"].(int)
+		if !ok {
+			h.Error(w, r, errors.New("invalid session value"), "")
+			return
+		}
+
+		p, ok := h.NewPage(w, r, "BoilerMake - Raffle")
+		if !ok {
+			h.Error(w, r, errors.New("creating page failed"), "")
+			return
+		}
+
+		user, err := h.ApplicationService.GetByUserID(id)
+		if err != nil { // TODO: Make check that user is checked-in here
```

Comment:
Instead of checking for checked-in, check to make sure user has applied. If not redirect to dashboard when message